galasa-managers-parent/galasa-managers-visualisation-parent/dev.galasa.elasticlog.manager/src/main/java/dev/galasa/elasticlog/internal/ElasticLogManagerImpl.java

```diff
@@ -0,0 +1,227 @@
+    private IConfigurationPropertyStoreService  cps;
+    private IConfidentialTextService			ctf;
+
+    private IHttpManagerSpi        		        httpManager;
+    private IHttpClient							client;
+    
+    private Gson								gson;
+
+    private HashMap<String, Object>				runProperties   = new HashMap<String, Object>(); 
+
+    /**
+     * Initialies the ElasticLogManager, adding the requirement of the HttpManager
+     *  
+     * @param IFramework - the galasa framework
+     * @param List<IManager> - list of all the managers
+     * @param List<Imanager> - list of all the active managers
+     * @param Class<?> - the test class
+     * @throws ManagerException
+     */
+    @Override
+    public void initialise(@NotNull IFramework framework, @NotNull List<IManager> allManagers,
+            @NotNull List<IManager> activeManagers, @NotNull Class<?> testClass) throws ManagerException {
+        super.initialise(framework, allManagers, activeManagers, testClass);
+
+        youAreRequired(allManagers, activeManagers);
+
+        try {
+            this.framework = framework;
+            this.cps = framework.getConfigurationPropertyService(NAMESPACE);
+            this.ctf = framework.getConfidentialTextService();
+            ElasticLogPropertiesSingleton.setCps(this.cps);
+        } catch (Exception e) {
+            throw new ElasticLogManagerException("Unable to request framework services", e);
+        }
+    }
+    
+	/**
+	 * Makes sure that the elastic log manager is added to the list of active managers, and adds the dependency on http manager.
+	 * 
+	 * @param List<IManager> - list of all the managers
+	 * @param List<IManager> - list of the active managers
+	 * @throws ManagerException
+	 */
+    @Override
+    public void youAreRequired(@NotNull List<IManager> allManagers, @NotNull List<IManager> activeManagers)
+            throws ManagerException {
+        if (activeManagers.contains(this))
+            return;
+        activeManagers.add(this);
+
+        httpManager = addDependentManager(allManagers, activeManagers, IHttpManagerSpi.class);
+    }
+
+    /**
+     * Provision build step, build the http client and gson object used by the manager 
+     * 
+     * @throws ManagerException
+     * @throws ResourceUnavailableException
+     */
+    @Override
+    public void provisionBuild() throws ManagerException, ResourceUnavailableException {
+        this.client = this.httpManager.newHttpClient();
+        this.gson = new GsonBuilder().setDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSS'Z'").create();
+        logger.info("ElasticLog Clients Initialised");
+    }
+
+    /**
+     * Test class result step, build and send the document request
+     * 
+     * @throws ManagerException
+     */
+    @Override
+	  	public void testClassResult(@NotNull String finalResult, Throwable finalException) throws ManagerException {
+	    //Record test information
+		this.runProperties.put("testCase", this.framework.getTestRun().getTestClassName());
```

Comment:
Similar to the PME manager in JAT,  you are going to need to ask each Manager for build information.
Create a new interface in framework,   search the managers for those that implement the interface and call them.